UnityProject/Assets/Scripts/Health/RespiratoryHealth/RespiratorySystem.cs

```diff
@@ -224,14 +298,48 @@ private float HandleBreathing(GasMix gasMix)
 	/// handled with interfaces we can implement so different species react differently.
 	/// </summary>
 	/// <param name="plasmaAmount"></param>
-	private void HandleBreathingPlasma(float plasmaAmount)
+	private float HandleBreathingPlasma(GasMix gasMix)
 	{
+		float plasmaPressure = gasMix.GetPressure(Gas.Plasma);
+
+		float plasmaConsumed = 0;
+
+		plasmaConsumed = gasMix.GetMoles(Gas.Plasma) * AtmosConstants.BREATH_VOLUME;
+
+		// the player is wearing a gas mask and there are less than 25 moles of plasma in the air
+		if (gameObject.Player() != null && gasMix.GetMoles(Gas.Plasma) < 25)
+		{
+			if (gameObject.Player().Script.Equipment.GetClothingItem(NamedSlot.mask).gameObject.GetComponent<ItemAttributesV2>() != null)
+			{
+				if (gameObject.Player().Script.Equipment.GetClothingItem(NamedSlot.mask).gameObject.GetComponent<ItemAttributesV2>().HasTrait(CommonTraits.Instance.GasMask))
+				{
+					if (DMMath.Prob(90))
```

Comment:
extract from here into its own method so you can reuse. Extract the ifs into a single bool property so you can simply do
```cs
if (isWearingGasMask)
{
    HandleGasMask(gasMix); 
}
```